app/src/test/java/org/oppia/android/app/parser/StringToFractionParserTest.kt

```diff
@@ -73,99 +73,217 @@ class StringToFractionParserTest {
   }
 
   @Test
-  fun testParser_submitTimeError_returnsValid() {
+  fun testParser_submitTimeError_regularFraction_returnsValid() {
     val error = stringToFractionParser.getSubmitTimeError("1/2")
-      .getErrorMessageFromStringRes(context)
-    assertThat(error).isEqualTo(null)
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.VALID)
   }
 
   @Test
-  fun testParser_submitTimeError_extraSpaces_returnsValid() {
-    val error = stringToFractionParser.getSubmitTimeError(" 1   / 2 ")
-      .getErrorMessageFromStringRes(context)
-    assertThat(error).isEqualTo(null)
+  fun testParser_submitTimeError_regularNegativeFractionWithExtraSpaces_returnsValid() {
+    val error = stringToFractionParser.getSubmitTimeError(" -1   / 2 ")
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.VALID)
   }
 
   @Test
-  fun testParser_submitTimeError_tooLong_returnsNumberTooLong() {
+  fun testParser_submitTimeError_atLengthLimit_returnsValid() {
+    val error = stringToFractionParser.getSubmitTimeError("1234567/1234567")
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.VALID)
+  }
+
+  @Test
+  fun testParser_submitTimeError_wholeNumber_returnsValid() {
+    val error = stringToFractionParser.getSubmitTimeError("888")
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.VALID)
+  }
+
+  @Test
+  fun testParser_submitTimeError_wholeNegativeNumber_returnsValid() {
+    val error = stringToFractionParser.getSubmitTimeError("-777")
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.VALID)
+  }
+
+  @Test
+  fun testParser_submitTimeError_mixedNumber_returnsValid() {
+    val error = stringToFractionParser.getSubmitTimeError("11 22/33")
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.VALID)
+  }
+
+  @Test
+  fun testParser_submitTimeError_returnsNumberTooLong() {
     val error = stringToFractionParser.getSubmitTimeError("0123456789")
-      .getErrorMessageFromStringRes(context)
-    assertThat(error)
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.NUMBER_TOO_LONG)
+
+    val errorMessage = error.getErrorMessageFromStringRes(context)
```

Comment:
Suggest moving these to their own tests (& also including a test for VALID). For example: ``testFractionParsingError_numberTooLong_hasRelevantErrorMessage`` or something similar.

Reply:
Done.